File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/monitoring/model/StageSnapshot.java
package com.study.webflux.rag.domain.monitoring.model;

import java.time.Instant;
import java.util.Map;

public record StageSnapshot(
	DialoguePipelineStage stage,
	StageStatus status,
	Instant startedAt,
	Instant finishedAt,
	long durationMillis,
	Map<String, Object> attributes
) {
	public StageSnapshot {
		attributes = attributes == null ? Map.of() : Map.copyOf(attributes);
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/monitoring/model/StageSnapshot.kt
package com.miyou.app.domain.monitoring.model

import java.time.Instant

data class StageSnapshot(
    val stage: DialoguePipelineStage,
    val status: StageStatus,
    val startedAt: Instant?,
    val finishedAt: Instant?,
    val durationMillis: Long,
    val attributes: Map<String, Any?>,
) {
    init {
        requireNotNull(attributes) { "attributes must not be null" }
    }
}
